src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

```diff
@@ -1390,11 +1390,12 @@ internal void WriteStringToListener(EventListener listener, string msg, SessionM
                 }
                 else
                 {
-                    List<object> arg = new List<object>();
-                    arg.Add(msg);
                     EventWrittenEventArgs eventCallbackArgs = new EventWrittenEventArgs(this);
                     eventCallbackArgs.EventId = 0;
-                    eventCallbackArgs.Payload = new ReadOnlyCollection<object>(arg);
+                    eventCallbackArgs.Message = msg;
+                    eventCallbackArgs.Payload = new ReadOnlyCollection<object>(new List<object>() { msg });
+                    eventCallbackArgs.PayloadNames = new ReadOnlyCollection<string>(new List<string> { "message" });
```

Comment:
Could arrays be used instead of `List<T>`?

```c#
eventCallbackArgs.Payload = new ReadOnlyCollection<object>(new object[] { msg });
eventCallbackArgs.PayloadNames = new ReadOnlyCollection<string>(new string[] { "message" });
```